### PR TITLE
feat(select): add js file - FRONT-1142

### PR DESCRIPTION
**Pull request by kalinchernev**
Preview available at https://select-multiple-dev--europa-component-library.netlify.app

### CODE REVIEW

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: small typo: defalt

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: this parameter seems to be a css prefix rather than a selector (we try to avoid to use css classes for selectors; data attributes are used instead when needed)
I'm not sure this parameter has to be exposed, as it won't be displayed correctly if another value is used here.

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: we should use a dedicated css class to hide the element; adding `display: none` would prevent data from being sent to the form.
Ideally this class should be added to the parent container of the select, so that I could remove extra margin too (something like `.ecl-select__container--hidden`)

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: I will provide the text for select all in a data attribute `data-ecl-select-all`

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: we may have conflict for ids if there are several multi-select on the page
I'd suggest using the id of the initial select element as prefix

**Review by emeryro — COMMENTED** [review]
Looks good so far!
The generated markup seems to be correct, which was the main challenge here.

I noted a few next steps to be done on both side:
on css/html side:
- hide select without display none
- remove extra spacing on select
- set position absolute for overlay
- add data for select all data-ecl-select-all

on js side:
- show/hide overlay on click
- handle select all
- add value to the select element when ticking a box

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: shoudn't it be selectContainer instead of searchContainer?

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: currently, this event is triggered when clicking a checkbox, so it is not possible to select mutliple ones without reopening the overlay every time.
I guess that putting the click event on the first ecl-select__container would be better

Note: there is a small video in a comment here, which shows expected behavior: https://citnet.tech.ec.europa.eu/CITnet/confluence/pages/viewpage.action?spaceKey=NEXTEUROPA&title=Selects+%28droplist%29+Multi-select+Specs

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js:340]: improvement for the select all: when the select all checkbox is checked, if user uncheck another checkbox, the select all checkbox should also becomes unchecked

**Review by emeryro — COMMENTED** [review]
thanks for the updates. We are getting closer to completion
I found a few points below that could be updated/fixed.

Apart of that, I think that the remaining points are:
- try to fix errors on IE
- handle search. For this point, I may see with COMM if it could be added on a second phase (if they want to have a running version of the component earlier)

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (kalinchernev) [reply src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: I double-checked, the markup at the moment is matching the one from the other pull request with [reference markup](https://5ecb7eefd2193ec6ca8e002c--europa-component-library.netlify.app/playground/ec/iframe.html?id=components-forms-select--multiple) where `ecl-select__multiple-dropdown` contains search input and checkboxes which is named searchContainer in the JS implementation

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (kalinchernev) [reply src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js:340]: Nice suggestion, thanks! Implemented.

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: here we should toggle display instead of visibility (`display: none` or `display: flex`).
As the checkboxes are not used in the form itself, it can be done, and it will improve the design

File: src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js]: as a small improvement, we may ignore font case (in the example, searching for "bel" won't keep option "Belgium" enabled)

**Review by emeryro — COMMENTED** [review]
Seems good, thanks
I assume that the search will be discussed more in detail when needed.
I just spotted two small improvements below, with that I think we will be able to ask for a first validation

**Review by emeryro — APPROVED** [review]
To me, the implementation is good.
Let's wait for the validation from COMM before we merge it

**Comment by kalinchernev** [comment]
Merging after confirmation in Jira

**Pull request merged**
